chia-wallet/src/derive_synthetic.rs

```diff
@@ -0,0 +1,125 @@
+use chia_bls::{PublicKey, SecretKey};
+use num_bigint::BigInt;
+use sha2::{digest::FixedOutput, Digest, Sha256};
+
+pub trait DeriveSynthetic {
+    fn derive_synthetic(&self, hidden_puzzle_hash: &[u8; 32]) -> Self;
+}
+
+impl DeriveSynthetic for PublicKey {
+    fn derive_synthetic(&self, hidden_puzzle_hash: &[u8; 32]) -> Self {
+        self + &synthetic_offset(self, hidden_puzzle_hash).public_key()
+    }
+}
+
+impl DeriveSynthetic for SecretKey {
+    fn derive_synthetic(&self, hidden_puzzle_hash: &[u8; 32]) -> Self {
+        self + &synthetic_offset(&self.public_key(), hidden_puzzle_hash)
+    }
+}
+
+pub fn hash_to_secret_key(bytes: [u8; 32]) -> SecretKey {
```

Comment:
is this different from `SecretKey::from_seed()`?
Perhaps the name of this function could highlight the difference, or the specific kind of key generation it uses